components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIAdminImpl.java

```diff
@@ -355,12 +358,36 @@ public long getTimestamp(String date) {
             keyManagerConfigurationsByTenant.addAll(keyManagerConfigurationsByOrganization);
         }
 
+        setAliasForTokenExchangeKeyManagers(keyManagerConfigurationsByTenant, tenantDomain);
+
         for (KeyManagerConfigurationDTO keyManagerConfigurationDTO : keyManagerConfigurationsByTenant) {
             decryptKeyManagerConfigurationValues(keyManagerConfigurationDTO);
         }
         return keyManagerConfigurationsByTenant;
     }
 
+    private void setAliasForTokenExchangeKeyManagers(List<KeyManagerConfigurationDTO> keyManagerConfigurationsByTenant,
+            String tenantDomain) throws APIManagementException {
+        for (KeyManagerConfigurationDTO keyManagerConfigurationDTO : keyManagerConfigurationsByTenant) {
+            if (org.apache.commons.lang3.StringUtils.equals(KeyManagerConfiguration.TokenType.EXCHANGED.toString(),
+                    keyManagerConfigurationDTO.getTokenType())) {
+                if (keyManagerConfigurationDTO.getExternalReferenceId() != null) {
+                    IdentityProvider identityProvider;
+                    try {
+                        identityProvider = IdentityProviderManager.getInstance()
+                                .getIdPByResourceId(keyManagerConfigurationDTO.getExternalReferenceId(), tenantDomain,
+                                        Boolean.FALSE);
+                    } catch (IdentityProviderManagementException e) {
+                        throw new APIManagementException("IdP retrieval failed. " + e.getMessage(),
```

Comment:
do we not pass e usually?

Reply:
Fixed via 446173df7a595cd00744cb964c8a6f7597b5d274